src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -196,14 +204,27 @@ private UpdateDocumentsCommand prepareRequestForAttachingDocumentToCase(CaseDocu
         List<DocumentUpdate> documentsList = new ArrayList<>();
         for (DocumentHashToken documentHashToken : caseDocumentsMetadata.getDocumentHashTokens()) {
 
+            Optional<ResponseEntity<StoredDocumentHalResource>> documentMetadata = Optional.ofNullable(
+                getDocumentMetadata(UUID.fromString(documentHashToken.getId())));
+
             if (documentHashToken.getHashToken() != null) {
                 String hashcodeFromStoredDocument = generateHashToken(UUID.fromString(documentHashToken.getId()));
```

Comment:
you might need to separate getDocumentMetadata(UUID) from generateHashToken(...) . Otherwise we endup calling twice DM store when hashToken is present scenario.